prow/plugins/config.go

```diff
@@ -1492,6 +1492,9 @@ func ResolveBugzillaOptions(parent, child BugzillaBranchOptions) BugzillaBranchO
 
 	if child.ExcludeDefaults == nil || !*child.ExcludeDefaults {
 		// populate with the parent
+		if parent.ExcludeDefaults != nil {
+			output.ExcludeDefaults = parent.ExcludeDefaults
```

Comment:
I don't believe this should be included. The updated test still passes without this line

Reply:
I added some more tests to more accurately illustrate the problem I'm facing:
```diff
diff --git a/prow/plugins/config_test.go b/prow/plugins/config_test.go
index 5fc7c569b2..f6abe7138d 100644
--- a/prow/plugins/config_test.go
+++ b/prow/plugins/config_test.go
@@ -797,7 +797,13 @@ orgs:
             status_after_merge: NOTABUG
           "my-special-branch":
             exclude_defaults: true
-            validate_by_default: false`
+            validate_by_default: false
+      another-repo:
+        branches:
+          "*":
+            exclude_defaults: true
+          "my-org-branch":
+            target_release: my-repo-branch`
        var config Bugzilla
        if err := yaml.Unmarshal([]byte(rawConfig), &config); err != nil {
                t.Fatalf("couldn't unmarshal config: %v", err)
@@ -857,6 +863,13 @@ orgs:
                        branch:   "my-special-branch",
                        expected: BugzillaBranchOptions{ValidateByDefault: &no, ExcludeDefaults: &yes},
                },
+               {
+                       name:     "exclude branch on repo cascades to branch config",
+                       org:      "my-org",
+                       repo:     "another-repo",
+                       branch:   "my-org-branch",
+                       expected: BugzillaBranchOptions{TargetRelease: &repoBranch, ExcludeDefaults: &yes},
+               },
        }
        for _, testCase := range testCases {
                t.Run(testCase.name, func(t *testing.T) {
@@ -941,6 +954,15 @@ orgs:
                                },
                        },
                },
+               {
+                       name: "excluded repo gets no defaults",
+                       org:  "my-org",
+                       repo: "another-repo",
+                       expected: map[string]BugzillaBranchOptions{
+                               "*":             {ExcludeDefaults: &yes},
+                               "my-org-branch": {ExcludeDefaults: &yes, TargetRelease: &repoBranch},
+                       },
+               },
        }
        for _, testCase := range repoTestCases {
                t.Run(testCase.name, func(t *testing.T) {
```

These fail without the above parent lines, the org defaults propagate into `my-org-branch` when I don't want them to.

Reply:
Ok, that makes sense. To prevent `exclude_defaults` in `org` or `global` from being passed to lower levels (and thus causing themselves to be excluded), can you add this change?

```
diff --git a/prow/plugins/config.go b/prow/plugins/config.go
index 5ef8bda48f..722cbd52d6 100644
--- a/prow/plugins/config.go
+++ b/prow/plugins/config.go
@@ -1649,12 +1649,16 @@ func (b *Bugzilla) OptionsForBranch(org, repo, branch string) BugzillaBranchOpti
        if !exists {
                return options
        }
+       // do not propagate exclude_defaults downwards
+       options.ExcludeDefaults = nil
        options = ResolveBugzillaOptions(options, OptionsForItem(branch, orgOptions.Default))
 
        repoOptions, exists := orgOptions.Repos[repo]
        if !exists {
                return options
        }
+       // do not propagate exclude_defaults downwards
+       options.ExcludeDefaults = nil
        options = ResolveBugzillaOptions(options, OptionsForItem(branch, repoOptions.Branches))
 
        return options
```

This will prevent downwards propagation while still allowing the wildcard repo branch (`*`) to set `exclude_defaults` for other branches in the repo.